### PR TITLE
perf: update bicep

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: graph-toolkit-contact-exporter/templates/azure/provision/simpleAuth.bicep [hunk 1]
```diff
@@ -3,9 +3,9 @@ param provisionParameters object
...
-var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}-simpleAuth-serverfarms'
-var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}-simpleAuth-webapp'
-var simpleAuthPackageUri = contains(provisionParameters, 'simpleAuthPackageUri') ? provisionParameters['simpleAuthPackageUri'] : 'https://github.com/OfficeDev/TeamsFx/releases/download/simpleauth@0.1.0/Microsoft.TeamsFx.SimpleAuth_0.1.0.zip'
+var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}simpleAuth'
+var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}simpleAuth'
+var simpelAuthPackageUri = contains(provisionParameters, 'simpleAuthPackageUri') ? provisionParameters['simpleAuthPackageUri'] : 'https://github.com/OfficeDev/TeamsFx/releases/download/simpleauth@0.1.0/Microsoft.TeamsFx.SimpleAuth_0.1.0.zip'
```
Comment (wenytang-ms) [inline graph-toolkit-contact-exporter/templates/azure/provision/simpleAuth.bicep:8]: please wait...

Comment (blackchoey) [inline graph-toolkit-contact-exporter/templates/azure/provision/simpleAuth.bicep:8]: please help change back to simpleAuthPackageUri

File: hello-world-bot/.fx/configs/azure.parameters.dev.json [hunk 2]
```diff
@@ -2,26 +2,16 @@
...
-    },
-    "m365TenantId": {
-      "value": "{{state.fx-resource-aad-app-for-teams.tenantId}}"
-    },
-    "m365OauthAuthorityHost": {
-      "value": "{{state.fx-resource-aad-app-for-teams.oauthHost}}"
-    }
+    "provisionParameters": {
+      "value": {
+          "resourceBaseName": "ssobotsampdev{{state.solution.resourceNameSuffix}}",
```
Comment (blackchoey) [inline hello-world-bot/.fx/configs/azure.parameters.dev.json:7]: update the resource base name?

File: hello-world-tab-with-backend/templates/azure/provision/simpleAuth.bicep [hunk 3]
```diff
@@ -3,8 +3,8 @@ param provisionParameters object
 param userAssignedIdentityId string
 var resourceBaseName = provisionParameters.resourceBaseName
 var sku = contains(provisionParameters, 'simpleAuthSku') ? provisionParameters['simpleAuthSku'] : 'F1'
-var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}-simpleAuth-serverfarms'
-var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}-simpleAuth-webapp'
+var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}simpleAuth'
+var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}simpleAuth'
 var simpelAuthPackageUri = contains(provisionParameters, 'simpleAuthPackageUri') ? provisionParameters['simpleAuthPackageUri'] : 'https://github.com/OfficeDev/TeamsFx/releases/download/simpleauth@0.1.0/Microsoft.TeamsFx.SimpleAuth_0.1.0.zip'
```
Comment (blackchoey) [inline hello-world-tab-with-backend/templates/azure/provision/simpleAuth.bicep:8]: please also help fix the typo: "simpelAuthPackageUri"

File: hello-world-tab/templates/azure/provision/simpleAuth.bicep [hunk 4]
```diff
@@ -3,8 +3,8 @@ param provisionParameters object
 param userAssignedIdentityId string
 var resourceBaseName = provisionParameters.resourceBaseName
 var sku = contains(provisionParameters, 'simpleAuthSku') ? provisionParameters['simpleAuthSku'] : 'F1'
-var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}-simpleAuth-serverfarms'
-var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}-simpleAuth-webapp'
+var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}simpleAuth'
+var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}simpleAuth'
 var simpelAuthPackageUri = contains(provisionParameters, 'simpleAuthPackageUri') ? provisionParameters['simpleAuthPackageUri'] : 'https://github.com/OfficeDev/TeamsFx/releases/download/simpleauth@0.1.0/Microsoft.TeamsFx.SimpleAuth_0.1.0.zip'
```
Comment (blackchoey) [inline hello-world-tab/templates/azure/provision/simpleAuth.bicep:8]: please also help fix the typo: "simpelAuthPackageUri" -> "simpleAuthPackageUri"

File: todo-list-with-Azure-backend-M365/templates/azure/provision/simpleAuth.bicep [hunk 5]
```diff
@@ -3,8 +3,8 @@ param provisionParameters object
 param userAssignedIdentityId string
 var resourceBaseName = provisionParameters.resourceBaseName
 var sku = contains(provisionParameters, 'simpleAuthSku') ? provisionParameters['simpleAuthSku'] : 'F1'
-var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}-simpleAuth-serverfarms'
-var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}-simpleAuth-webapp'
+var serverFarmsName = contains(provisionParameters, 'simpleAuthServerFarmsName') ? provisionParameters['simpleAuthServerFarmsName'] : '${resourceBaseName}simpleAuth'
+var webAppName = contains(provisionParameters, 'simpleAuthWebAppName') ? provisionParameters['simpleAuthWebAppName'] : '${resourceBaseName}simpleAuth'
 var simpelAuthPackageUri = contains(provisionParameters, 'simpleAuthPackageUri') ? provisionParameters['simpleAuthPackageUri'] : 'https://github.com/OfficeDev/TeamsFx/releases/download/simpleauth@0.1.0/Microsoft.TeamsFx.SimpleAuth_0.1.0.zip'
```
Comment (blackchoey) [inline todo-list-with-Azure-backend-M365/templates/azure/provision/simpleAuth.bicep:8]: please help fix the typo

File: graph-toolkit-contact-exporter/templates/azure/provision/simpleAuth.bicep (hunk 1 above)
Comment (KennethBWSong) [reply graph-toolkit-contact-exporter/templates/azure/provision/simpleAuth.bicep]: Fixed. Thanks!

Comment (KennethBWSong) [reply graph-toolkit-contact-exporter/templates/azure/provision/simpleAuth.bicep]: Fixed. Thanks!

File: hello-world-bot/.fx/configs/azure.parameters.dev.json (hunk 2 above)
Comment (KennethBWSong) [reply hello-world-bot/.fx/configs/azure.parameters.dev.json]: Fixed. Thanks!

File: hello-world-tab-with-backend/templates/azure/provision/simpleAuth.bicep (hunk 3 above)
Comment (KennethBWSong) [reply hello-world-tab-with-backend/templates/azure/provision/simpleAuth.bicep]: Fixed. Thanks!

File: hello-world-tab/templates/azure/provision/simpleAuth.bicep (hunk 4 above)
Comment (KennethBWSong) [reply hello-world-tab/templates/azure/provision/simpleAuth.bicep]: Fixed. Thanks!

File: todo-list-with-Azure-backend-M365/templates/azure/provision/simpleAuth.bicep (hunk 5 above)
Comment (KennethBWSong) [reply todo-list-with-Azure-backend-M365/templates/azure/provision/simpleAuth.bicep]: Fixed. Thanks!

**Review by blackchoey — APPROVED** [review]

**Review by wenytang-ms — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Review by dilin-MS — APPROVED** [review]

**Pull request merged**